sunpy/timeseries/timeseries_factory.py

```diff
@@ -531,6 +513,16 @@ def _check_registered_widgets(self, **kwargs):
         return WidgetType(data, meta, units, **kwargs)
 
 
+def _apply_result(data_header_pairs, filepaths, result):
```

Comment:
No really sure about this myself just the code was repeated 3/4 time in quick succession ...

Reply:
not sure about the name, but separating it out seems to make sense.